README.md

```diff
@@ -15,6 +47,507 @@ go get -u github.com/src-d/kallax/...
+
+## Model schema
+
+### Automatic schema generation and migrations
+
+Automatic `CREATE TABLE` for models and migrations is not yet supported, even though it will probably come in future releases.
+
+### Use schema
+
+A global variable `Schema` will be created in your `kallax.go`, that contains a field with the name of every of your models. Those are the schemas of your models. Each model schema contains all the fields of that model.
+
+So, to access the username field of the user model, it can be accessed as:
+
+```go
+Schema.User.Username
+```
+
+## Manipulate models
+
+For all of the following sections, we will assume we have a store `store` for our model's type.
+
+### Insert models
+
+To insert a model we just need to use the `Insert` method of the store and pass it a model. If the model does not have an ID, one will be assigned to it.
+
+```go
+user := NewUser("fancy_username", "super_secret_password", "foo@email.me")
+err := store.Insert(user)
+if err != nil {
+        // handle error
+}
+```
+
+If our model has relationships, they will be saved (**note:** saved as in insert or update) as well. The relationships of the relationships will not, though. Relationships are only saved with one level of depth.
+
+```go
+user := NewUser("foo")
+user.Posts = append(user.Posts, NewPost(user, "new post"))
+
+err := store.Insert(user)
+if err != nil {
+        // handle error
+}
+```
+
+If there are any relationships in the model, both the model and the relationships will be saved in a transaction and only succeed if all of them are saved correctly.
+
+### Update models
+
+To insert a model we just need to use the `Update` method of the store and pass it a model. It will return an error if the model was not already persisted or has not an ID.
+
+```go
+user := FindLast()
+rowsUpdated, err := store.Update(user)
+if err != nil {
+        // handle error
+}
+```
+
+By default, when a model is updated, all its fields are updated. You can also specify which fields to update passing them to update.
+
+```go
+rowsUpdated, err := store.Update(user, Schema.User.Username, Schema.User.Password)
+if err != nil {
+        // handle error
+}
+```
+
+If our model has relationships, they will be saved (**note:** saved as in insert or update) as well. The relationships of the relationships will not, though. Relationships are only saved with one level of depth.
+
+```go
+user := FindLastPoster()
+rowsUpdated, err := store.Update(user)
+if err != nil {
+        // handle error
+}
+```
+
+If there are any relationships in the model, both the model and the relationships will be saved in a transaction and only succeed if all of them are saved correctly.
+
+### Save models
+
+To save a model we just need to use the `Save` method of the store and pass it a model. It will update the model if it was already persisted or insert it otherwise.
```

Comment:
I'd explain that this method is just a shortcut to the proper `update` or `insert` depending on the previous state of the object, and I would explain no more. This way, we will avoid copy-paste in documentation, that can cause incoherence between the `save` and the main methods (example: it is not explained here that the save will fail if the object has not set its pk)